cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -98,6 +98,17 @@ Result<std::shared_ptr<SchemaManifest>> GetSchemaManifest(
   return manifest;
 }
 
+bool isNan(const Scalar& value) {
```

Comment:
Please, let's follow the coding conventions
```suggestion
bool IsNan(const Scalar& value) {
```

Reply:
Thanks for the review, apologies for missing that out. Made the change!